### PR TITLE
React index page

### PR DESCRIPTION
**Pull request by kwong402**
Alana:
- Changed Index to React Component
- Updated Browser Router in App.js
- Updated Shops Controller
- Updated the routes.rb to show index in api/vi/shops
Karen:
- Updated indents for Alana since her VSCode makes 4 spaces, but we still see 2

### CODE REVIEW

File: app/controllers/api/v1/shops_controller.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Api::V1::ShopsController < ApiController 
+  def index
+    render json: { shops: Shop.all }
```
Comment (nwalberts) [inline app/controllers/api/v1/shops_controller.rb:3]: this is fine for now, but its time to start making serializers for your endpoints. Your next (non-styling) PR should have a serializer

File: app/javascript/react/components/App.js [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
-    </BrowserRouter>
+    <div>
+      <BrowserRouter>
+        <Switch>
+          <Route exact path="/" component={ShopIndexPage} />
+          <Route exact path="/shops" component={ShopIndexPage} />
+          <Route exact path="/shops/:id" component={ShopShowPage} />
+        </Switch>
+      </BrowserRouter>
+    </div>
```
Comment (nwalberts) [inline app/javascript/react/components/App.js]: this div isnt doing anything. lets remove it!

File: app/javascript/react/components/ShopIndexPage.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  useEffect(() => {
+    showShops()
+  }, [])
+
+  const shopsList = getShops.map((shop) => {
+    return (
+      <li key={shop.id}>
+        <Link to={`/shops/${shop.id}`}>{shop.name}</Link>
+      </li>
```
Comment (nwalberts) [inline app/javascript/react/components/ShopIndexPage.js]: I know its a small element, but lets make this into a tile. React docs do say that you ideally shouldnt pass keys to elements, but instead as props to componente

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Comment by kwong402** [comment]
- Made minor fixes
- Made ShopTile component

**Comment by kwong402** [comment]
- Added back test for index page that we will want to delete after to make a controller test

**Pull request merged**